clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go

```diff
@@ -221,16 +221,26 @@ func (w *waitForControlledPodsRunningMeasurement) gather(c clientset.Interface,
 	w.handlingGroup.Wait()
 	w.lock.Lock()
 	defer w.lock.Unlock()
-	numberRunning := 0
+	var numberRunning, numberTerminated, numberUnknown int
 	for _, checker := range w.checkerMap {
-		if status := checker.getStatus(); status {
+		switch checker.getStatus() {
+		case running:
 			numberRunning++
+		case terminated:
+			numberTerminated++
+		default:
+			numberUnknown++
 		}
 	}
+	glog.Infof("%s: running %d, terminated %d, unknown: %d", w, numberRunning, numberTerminated, numberUnknown)
 	if desiredCount != numberRunning {
 		glog.Errorf("%s: incorrect objects number: %d/%d %ss are running with all pods", w, numberRunning, desiredCount, w.kind)
 		return fmt.Errorf("incorrect objects number")
 	}
+	if numberUnknown > 0 {
+		glog.Errorf("%s: unknown status for %d %ss", w, numberUnknown, w.kind)
+		return fmt.Errorf("unknown objects statuses")
```

Comment:
I'm not sure it's really unknown - usuallly, it would mean that simply we didn't yet finish reconciling. Is that right?

Reply:
It's true. However the reason of operation not being finished is unknown. That's why I though "status unknown" is a good name.
If this error is returned, there definitely will be some logs explaining the situation.

Reply:
Do we want to have more information in the error message or in error log? What kind of information should be added?
For me this error is only a signal that something went wrong and I expect user to look into log file to see more details - that's why I tried to keep error message simple.

Reply:
That's a bad way - error should tell you what happened. Users may look into logs to get information "why", but not to look "what really happened".

Things that are part of log (e.g. what object that was) should be part of error.

In fact (as I mentioned sometime in the past), I'm not convinced our logs are what we really need. We should probably do the audit of them soon.

Reply:
Ok.
I added the list of errors for the objects with unknown status. This errors contain object names with error descriptions.